src/components/Subscribe/Subscribe.tsx

```diff
@@ -52,8 +68,15 @@ const Subscribe = ({
   inputLabelText = 'Email*',
   inputPlaceholder = 'example@email.com',
   title = 'Subscribe to Newsletter',
+  loadingText = 'Loading...',
+  invalidText = '',
+  subscriptionState = null,
+  onInputChange = () => ({}),
   ...props
 }: SubscribeProps) => {
+  const isInvalid = subscriptionState === 'invalid';
+  const isLoading = subscriptionState === 'loading';
+
   return (
     <Element
```

Comment:
I would add the `onSubmit` handler to the parent form instead of the button's `onClick` using the form submission behavior

Reply:
It's what I did, just left onClick for button just in case